pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -55,6 +55,8 @@ const (
 	defaultIOThread        = uint(1)
 	EFIPath                = "/usr/share/OVMF/OVMF_CODE.fd"
 	EFIVarsPath            = "/usr/share/OVMF/OVMF_VARS.fd"
+	EFIPathSecureBoot      = "/usr/share/OVMF/OVMF_CODE.secboot.fd"
```

Comment:
I wonder if we need to make this configurable ... I think that I have also read somewhere that libvirt may know these locations on each OS it is running on, but that may be wrong.

Reply:
About the functional test, I started writing one but, as you predicted, `NewRandomVMIWithEFIBootloader()` starts an unsigned binary and fails.
Locally I've been testing with the Fedora 31 live CD / installed, that boots just fine.

About the paths, they can be configured in /etc/libvirt/qemu.conf, but I don't think we can count on it. We probably want to be able to switch between regular and secboot anyway.
However, there is indeed a recent libvirt option that automatically sets paths and stuff (including SMM! yikes). I tried adding support for it but miserably failed... This is where I'm at:
https://github.com/jean-edouard/kubevirt/commit/92a9abec6b541188448495d1b42912de909abeeb

Reply:
Do we need a way to set the paths? I guess we should at least be able to successfully boot with the right pathes on centos/fedora/rhel.

Reply:
I'd say it's fine to hardcode the paths since libvirt does too.
Before v.6.0.0, libvirt had a (compile-time) exception for Fedora. Seems like before v31, Fedora used different paths than the rest of the world.
The exception was removed there:
https://github.com/libvirt/libvirt/commit/5a5e92000d12a671f491c5fb90677f63b1ae7e75#diff-c49a1d135157a3c9aac84639bc92a285

Maybe we should work with libvirt to improve their logic for choosing the right firmware files, ultimately allowing us to remove those hardcoded paths from our codebase.

Reply:
I'd say hardcoded paths are fine as a temporary measure, until libvirt gets upgraded to a better version.

From v6.0.0 forward, the paths are already fixed, but it seems that they'll always use the secboot images if present, which I don't think is what we want.

(
Before v.6.0.0, libvirt had a (compile-time) exception for Fedora. Seems like before v31, Fedora used different paths than the rest of the world.
The exception was removed there:
https://github.com/libvirt/libvirt/commit/5a5e92000d12a671f491c5fb90677f63b1ae7e75#diff-c49a1d135157a3c9aac84639bc92a285
)

Maybe we should work with them to implement a switch that picks the secboot files only if secureboot was requested, ultimately lifting the need for hardcoded paths in our codebase.